src/connector/src/sink/encoder/json.rs

```diff
@@ -18,29 +18,31 @@ use chrono::{Datelike, Timelike};
 use risingwave_common::array::{ArrayError, ArrayResult};
 use risingwave_common::catalog::{Field, Schema};
 use risingwave_common::row::Row;
-use risingwave_common::types::{DataType, DatumRef, ScalarRefImpl, ToText};
+use risingwave_common::types::{DataType, DatumRef, Decimal, ScalarRefImpl, ToText};
 use risingwave_common::util::iter_util::ZipEqDebug;
 use serde_json::{json, Map, Value};
 
-use super::{Result, RowEncoder, SerTo, TimestampHandlingMode};
+use super::{CustomJsonType, Result, RowEncoder, SerTo, TimestampHandlingMode};
 use crate::sink::SinkError;
-
 pub struct JsonEncoder<'a> {
     schema: &'a Schema,
     col_indices: Option<&'a [usize]>,
     timestamp_handling_mode: TimestampHandlingMode,
+    custom_json_type: CustomJsonType,
 }
 
 impl<'a> JsonEncoder<'a> {
     pub fn new(
         schema: &'a Schema,
         col_indices: Option<&'a [usize]>,
         timestamp_handling_mode: TimestampHandlingMode,
+        custom_json_type: CustomJsonType,
```

Comment:
Since custom_json_type is only used in doris sink, I think it is better to have two constructors: one that provides a custom_json_type and one that doesn't provide it. Non-doris sink will use the later one to construct JsonEncoder